### DISCUSSION TITLE
Show extension in file loaders

### DISCUSSION THREAD

**Discussion by 38github** [rerdavies/pipedal#210] [Ideas]
I am using a plugin that can load both .nam and .wav files and Pipedal puts all these together. Do you have any idea how to differenciate these from each other in the listings or may an icon to the left of them, like a square, with the extension in it be something that is doable?

**Comment by maarthome** [comment]
Hi @38github,

Which plugin is that? 
With AIDA-X I can load also both... model and IR files. 
![image](https://github.com/user-attachments/assets/3beba583-0caf-4112-89da-2979513308a8)
For that I created separate linked directories:
![image](https://github.com/user-attachments/assets/92f66d78-5937-4095-93de-586e6acfe8bd)
It becomes a little bit 'messy' but at least not everything is in the same directory when selecting.

Cheers,
Maarten

**Reply by 38github** [reply]
I use Ratatouille. Maybe I have missed that I can navigate through different folders. I will look into that. Thank you!

**Comment by maarthome** [comment]
Hi, I is not navigating, it is creating links to directoriess where the model and IR files are located using symbolic or hard links....

Cheers,
Maarten

**Reply by maarthome** [reply]
Here I am again,

i see that ratatouille puts everything in one directory....
But in the selection window it filters on extension.... 
That is workable?

Cheers,

Maarten

**Reply by 38github** [reply]
Yes, filtering would be really good! Tack!

**Reply by rerdavies** [reply]
@maarthome:

The directories you are simlinking to... Where are they located, and who/what created them? Did you create separate directories for IR files and models? Are there separate directories for .nam, .json, and .aidx files?

**Reply by rerdavies** [reply]
@maarthome 

Nvm. I see the sources for your simlinks. Could you do me a very great favor, and run the following commands for me.

```
      grep -r --include "*.ttl"  ROOT_DIRECTORY_OF_YOUR_LV2_PLUGINS
```

```
     find ROOT_DIRECTORY_OF_YOUR_MEDIA_FILES -type d 
```

**Comment by rerdavies** [comment]
Looking at both plugins now. Doing this right is going to be complicated. :-/ 

Does anyone have other examples of plugins that use file path properties? 

The outcome is really going to depend on how consistently plugins indicate what file types they expect using an extended attribute from the MOD project (mod:fileType).  Ratatouille, apparent does use the mod:fileType extension.  The current version of Mike's NAM also uses the mod:fileType extension, although previous versions did not.  I'll get back to you with a draft spec for a better implementation to see what you think. 

Likely features: 

-   shared directories for audio files/
-  shared directories for .nam/.nammodel files.  
-  per-plugin Isolated directories for .json files (Toom ML/Proteus).
-  per-plugin isolated directories for everything esle.
-  File extensions displayed.
-  Filtering by extension,
-  A migration tool for existing media files. presets and banks (with the ability to revert).

Per-plugin isolated directories for everything else.

**Reply by rerdavies** [reply]
Just as an aside, to perhaps provide useful context in this conversation, one of the features that recently went in to PiPedal (actually v1.3.53, but first announced in v1.3.54)  `:-/`

No more splatting of huge number of files into the root upload directory. Uploaded .zip files always get a containing directory (either the root directory of the zip file, or the name of the zip file itself.

That doesn't in any way fix the problems you're having. But it at least makes it easier to not commingle .nam and .wav files in the same directory, which reduces your suffering slightly.

**Comment by maarthome** [comment]
Hi Robin,

grep -r --include "*.ttl"  /usr/lib/lv2 -> did not result in anything....
so I did....  grep -r --include "*.ttl" .   (while in .usr/lib/lv2)

[Pipedal.zip](https://github.com/user-attachments/files/17324007/Pipedal.zip)

Hope it is of some use,

Maarten

**Reply by rerdavies** [reply]
Exactly what I needed. Thanks.
